src/main/java/org/onedatashare/server/controller/CredController.java

```diff
@@ -35,4 +37,10 @@
     return userService.getCredentials(headers.getFirst(ODSConstants.COOKIE));
   }
 
+  @PostMapping("/saveCredentials")
+  public Mono<Map<UUID, Credential>> saveCredentials(@RequestHeader HttpHeaders headers, @RequestBody List<OAuthCredential> credentials){
```

Comment:
We do not need to send the credentials map back to the front end in this case. In my opinion the return type should be a void or a boolean to indicate if the operation was successful.